pkgs/by-name/op/opengamepadui/package.nix

```diff
@@ -0,0 +1,122 @@
+{
+  alsa-lib,
+  autoPatchelfHook,
+  cargo,
+  dbus,
+  fetchFromGitHub,
+  gamescope,
+  gitUpdater,
+  godot_4,
+  godot_4-export-templates,
+  hwdata,
+  lib,
+  libGL,
+  libpulseaudio,
+  mesa-demos,
+  pkg-config,
+  rustPlatform,
+  stdenv,
+  udev,
+  upower,
+  vulkan-loader,
+  xorg,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "opengamepadui";
+  version = "0.35.1";
+  withDebug = false;
```

Comment:
I'm not sure if this is ideal. I personally would prefer overriding via override and not overrideAttrs

Reply:
Should I change this back to be an argument to the derivation? I'm not sure which pattern is preferred.

Reply:
IIUC `overrideAttrs` is preferred. The issue with having it as a function parameter is that if an overlay adds a package called, e.g., `withDebug`, this can cause some weird issues. This _has_ happened before, although I don't have a reference to it.

Reply:
I don't see discussion about this anywhere. Besides, if *that's* the issue, then a lot of things in Nixpkgs would be royally screwed. (See, for example, ffmpeg.) AFAICT the current standard is still to use a lambda parameter.

Reply:
I have updated `package.nix` to use the lambda parameter pattern. If there are examples or consensus on moving away from that pattern, I can revert this change if needed.